AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -63,6 +66,57 @@ - (BOOL)deleteUserStorageWithAccountId:(NSString *)accountId {
+                             readOptions:(MSReadOptions *)__unused readOptions {
+  NSString *selectionQuery =
+      [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\" AND \"%@\" = \"%@\"", kMSAppDocumentTableName,
+                                 kMSPartitionColumnName, partition, kMSDocumentIdColumnName, documentId];
+  NSArray *result = [self.dbStorage executeSelectionQuery:selectionQuery];
+
+  // Return an error if the entry could not be found in the database.
+  if (result.count == 0) {
+    NSString *errorMessage = [NSString
+        stringWithFormat:@"Unable to find document in local database with partition key '%@' and document ID '%@'", partition, documentId];
+    MSLogWarning([MSDataStore logTag], @"%@", errorMessage);
+    NSError *error = [[NSError alloc] initWithDomain:kMSACDataStoreErrorDomain
+                                                code:MSACDataStoreErrorLocalDocumentNotFound
+                                            userInfo:@{NSLocalizedDescriptionKey : errorMessage}];
+    return [[MSDocumentWrapper alloc] initWithError:error documentId:documentId];
+  }
+
+  // If the document is expired, return an error and delete it.
+  NSDate *expirationTime = [MSUtility dateFromISO8601:result[0][self.expirationTimeColumnIndex]];
+  NSDate *currentDate = [NSDate date];
+  if ([expirationTime laterDate:currentDate] == currentDate) {
+    NSString *errorMessage = [NSString stringWithFormat:@"Local document with partition key '%@' and document ID '%@' expired at %@",
+                                                        partition, documentId, expirationTime];
+    MSLogWarning([MSDataStore logTag], @"%@", errorMessage);
+    NSError *error = [[NSError alloc] initWithDomain:kMSACDataStoreErrorDomain
+                                                code:MSACDataStoreErrorLocalDocumentExpired
+                                            userInfo:@{NSLocalizedDescriptionKey : errorMessage}];
+    [self deleteDocumentWithPartition:partition documentId:documentId];
+    return [[MSDocumentWrapper alloc] initWithError:error documentId:documentId];
+  }
+
+  // Deserialize.
+  NSString *jsonString = result[0][self.documentColumnIndex];
+  NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+  NSDate *lastUpdatedDate = [MSUtility dateFromISO8601:result[0][self.operationTimeColumnIndex]];
+  return [MSDocumentUtils documentWrapperFromDocumentData:jsonData
+                                             documentType:documentType
+                                                     eTag:result[0][self.eTagColumnIndex]
+                                          lastUpdatedDate:lastUpdatedDate
+                                                partition:partition
+                                               documentId:documentId];
+}
+
+- (void)deleteDocumentWithPartition:(NSString *)__unused partition documentId:(NSString *)__unused documentId {
+
+  // TODO implement this.
```

Comment:
It is being implemented in https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1475